cli/src/torii/routing.rs

```diff
@@ -170,32 +170,44 @@ pub(crate) async fn handle_queries(
 }
 
 fn apply_sorting_and_pagination(
-    mut vec_of_val: Vec<Value>,
+    vec_of_val: Vec<Value>,
     sorting: &Sorting,
     pagination: Pagination,
 ) -> Vec<Value> {
     if let Some(ref key) = sorting.sort_by_metadata_key {
-        let f = |value1: &Value| {
-            if let Value::Numeric(NumericValue::U128(num)) = value1 {
-                *num
-            } else {
-                0
-            }
-        };
-
-        vec_of_val.sort_by_key(|value0| match value0 {
-            Value::Identifiable(IdentifiableBox::Asset(asset)) => match asset.value() {
-                AssetValue::Store(store) => store.get(key).map_or(0, f),
-                _ => 0,
+        let mut pairs: Vec<(Option<Value>, Value)> = vec_of_val
+            .into_iter()
+            .map(|value| {
+                let key = match &value {
+                    Value::Identifiable(IdentifiableBox::Asset(asset)) => match asset.value() {
```

Comment:
This `match` arm shadows numerical values which cannot be sorted by metadata. Have you tried a more specific pattern before a more general one? 

```suggestion
                    Value::Identifiable(IdentifiableBox::Asset(AssetValue(asset))) => asset.get(key).cloned()
```

Reply:
Opened [an issue](https://github.com/hyperledger/iroha/issues/3139) about sorting numeric values.